dev-packages/application-package/src/application-package.ts

```diff
@@ -87,7 +88,7 @@ export class ApplicationPackage {
             theia.target = defaultTarget;
         }
 
-        return this._props = { ...ApplicationProps.DEFAULT, ...theia };
+        return this._props = merge(ApplicationProps.DEFAULT, theia);
```

Comment:
Does it mutate and return `ApplicationProps.DEFAULT`? Or it creates a new instance and merge defaults into it?

Reply:
It creates a new object. Check the [docs](https://www.npmjs.com/package/deepmerge#mergex-y-options):

> Merging creates a new object, so that neither x or y is modified.